crates/mun_codegen/src/value/string.rs

```diff
@@ -0,0 +1,56 @@
+use super::{AsValue, Global, IrValueContext, TransparentValue, Value};
+use crate::value::AddressableType;
+use std::ffi::{CStr, CString};
+
+/// Enables internalizing certain data structures like strings.
+pub trait CanInternalize {
+    type Type;
+
+    /// Internalizes the instance into a global value.
+    fn intern<S: AsRef<str>>(&self, name: S, context: &IrValueContext) -> Global<Self::Type>;
+}
+
+impl CanInternalize for str {
+    type Type = String;
+
+    fn intern<S: AsRef<str>>(&self, name: S, context: &IrValueContext) -> Global<Self::Type> {
+        Global::from_raw(
+            self.as_bytes()
+                .as_value(context)
+                .into_const_private_global(name, context)
+                .value,
+        )
+    }
+}
+
+impl CanInternalize for CStr {
+    type Type = CString;
+
+    fn intern<S: AsRef<str>>(&self, name: S, context: &IrValueContext) -> Global<Self::Type> {
+        Global::from_raw(
+            self.to_bytes_with_nul()
+                .as_value(context)
+                .into_const_private_global(name, context)
+                .value,
+        )
+    }
+}
+
+impl TransparentValue for CString {
+    type Target = [u8];
+
+    fn as_target_value(&self, context: &IrValueContext) -> Value<Self::Target> {
+        self.as_bytes_with_nul().as_value(context)
+    }
+}
+
+impl TransparentValue for String {
+    type Target = [u8];
+
+    fn as_target_value(&self, context: &IrValueContext) -> Value<Self::Target> {
+        self.as_bytes().as_value(context)
```

Comment:
No, not at all. Why do you ask? 

For clarity. I added both conversions from CString and String because they have different behaviors (can contain 0's, encoding, and null terminator). In the ABI we have to use CStrings so there we use that. If however, we added a string type later on I imagine we want to stay close to rusts String type.

Reply:
I was thinking that our safe Inkwell wrapper would only be applicable to symbol generation. As that has to conform to the C ABI, I thought using the `String` and `str` types directly would potentially break that.